### ISSUE TITLE
Fix import / export settings

### ISSUE THREAD

**Issue by herrvigg** [herrvigg/qtranslate-x#20]
<a href="https://github.com/xhaleera"><img src="https://avatars0.githubusercontent.com/u/3322862?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [xhaleera](https://github.com/xhaleera)**
_Sunday Feb 15, 2015 at 11:03 GMT_
_Originally opened as https://github.com/qTranslate-Team/qtranslate-x/pull/20_

----

----

_**[xhaleera](https://github.com/xhaleera)** included the following code: https://github.com/qTranslate-Team/qtranslate-x/pull/20/commits_

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Sunday Feb 15, 2015 at 20:20 GMT_

----

Thank you, @xhaleera, I probably forgot to adjust this after deeper separating front/back files and actions.

Would not this be even better:

```
function qtranxf_migrate_plugin($plugin){
    $var=$plugin.'-migration';
    if(!isset($_POST[$var])) return;
    if($_POST[$var]=='none') return;
    if($_POST[$var]=='export')
        qtranxf_saveConfig();
    $f='qtranxf_migrate_'.$_POST[$var].'_'.str_replace('-','_',$plugin);
    $f();
    if($_POST[$var]=='import')
        qtranxf_loadConfig();
}

function qtranxf_migrate_plugins()
{
    if(!current_user_can('manage_options')) return;
    qtranxf_migrate_plugin('mqtranslate');
    qtranxf_migrate_plugin('qtranslate-xp');
    //qtranxf_migrate_plugin('ztranslate');//ok same db
}
add_action('qtranslate_init_begin','qtranxf_migrate_plugins',11);
```

I also added `if(!current_user_can('manage_options')) return;`.

**Comment by herrvigg** [comment]
<a href="https://github.com/xhaleera"><img src="https://avatars0.githubusercontent.com/u/3322862?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [xhaleera](https://github.com/xhaleera)**
_Sunday Feb 15, 2015 at 22:38 GMT_

----

I don't exactly understand why you need the saveConfig() part, but the rest seems ok.

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Sunday Feb 15, 2015 at 23:06 GMT_

----

Because potentially some other options could have been set at the same time, which are not stored yet. That was my thinking, but I might be wrong, maybe they are always saved already at this point. It has been a while I was working on this. If you already checked that this is excessive, I will remove it.

**Comment by herrvigg** [comment]
<a href="https://github.com/xhaleera"><img src="https://avatars0.githubusercontent.com/u/3322862?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [xhaleera](https://github.com/xhaleera)**
_Monday Feb 16, 2015 at 01:17 GMT_

----

No, I've not checked it. You should keep it.

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Monday Feb 16, 2015 at 20:15 GMT_

----

Ok, then it is done, thank you very much, Christophe!

**Comment by x06designs** [comment]
Repro session notes (env: WP local, classic editor, ACF 6.8.9, module `acf` active, `qtranslate_module_acf` unset so all `standard_fields` default to true).

Confirmed so far:
- A field group with a plain `text` field and a `repeater` whose sub-field is `text` was imported. The plain text field **is** hooked on load: it carries `qtranxs-translatable` and gets its LSB.
- The scan in `js/acf/load.js` runs exactly once, inside `wp.hooks.addAction('qtranx.load', ...)`. There is no `acf.addAction('append', ...)`, so a row added after load is never scanned. The commented-out block at `js/acf/load.js:95-108` (`// TODO: fix RepeaterField #882`) only ever addressed row *removal*, never addition.

Separate defect found while probing, in the same file:
`qTranx.hooks.hasContentHook(this)` is passed a **DOM element** at all three call sites, but `hasContentHook(id)` (`js/core/hooks/handlers.js:52`) indexes `_contentHooks` by the element **id** string. The guard therefore always evaluates false. Harmless today only because `addContentHook()` re-checks `_contentHooks[inputField.id]` itself (`handlers.js:118`) — but the fix for this issue relies on that guard to keep a re-scan idempotent, so it has to be corrected in the same change (pass `this.id`).

Still to do: click "Add Row", confirm the new sub-field gets no LSB, then hook `acf.addAction('append', ...)` and re-run the scan scoped to the appended subtree. `acf.findFields()` could not be confirmed to accept a `parent` option in 6.8.9, so scope with jQuery over the appended `$el` instead.
